api/src/utils/apply-query.ts

```diff
@@ -357,7 +357,13 @@ export function applyFilter(
 
 					if (!columnName) continue;
 
-					applyFilterToQuery(columnName, filterOperator, filterValue, logical);
+					if (relation && relation.related_collection) {
```

Comment:
```suggestion
					if (relation?.related_collection) {
```

Super minor and not really important change 😄